exporter/azuremonitorexporter/log_to_envelope.go

```diff
@@ -55,11 +53,8 @@ func (packer *logPacker) LogRecordToEnvelope(logRecord plog.LogRecord) *contract
 
 	messageData.Message = logRecord.Body().StringVal()
 
-	hexTraceID := logRecord.TraceID().HexString()
-	messageData.Properties[traceIDTag] = hexTraceID
-	envelope.Tags[contracts.OperationId] = hexTraceID
-
-	messageData.Properties[spanIDTag] = logRecord.SpanID().HexString()
+	envelope.Tags[contracts.OperationId] = logRecord.TraceID().HexString()
+	envelope.Tags[contracts.OperationParentId] = logRecord.SpanID().HexString()
 
```

Comment:
The solution in the link seems incorrect. TraceId is available as operation id. There is well defined place to store trace id. CustomDimension is for anything custom, and traceid is not one of them.

Reply:
https://docs.microsoft.com/en-us/azure/azure-monitor/app/correlation#example

Reply:
1. The solution is correct. The logs (and traces) on [this screenshot](https://github.com/Azure-Samples/iotedge-logging-and-monitoring-solution/blob/main/docs/iot-distr-tracing-logs.png) were not exported by the Otel Collector and this exporter. They were exported directly to App Insights with this configuration:
```
 private ILogger GetLogger()
        {
            using var loggerFactory = LoggerFactory.Create(builder =>
            {
                builder
                    .AddFilter("Microsoft", LogLevel.Warning)
                    .AddFilter("System", LogLevel.Warning)
                    .AddSimpleConsole(options => options.IncludeScopes = true)
                    .AddApplicationInsights(GetEnvironmentVariable("INSTRUMENTATION_KEY"))
                    .Configure(c => c.ActivityTrackingOptions =
                        ActivityTrackingOptions.SpanId
                        | ActivityTrackingOptions.TraceId);
            });

            return loggerFactory.CreateLogger<IoTHubCatcher>();
        }
```

It uses [ActivityTrackingOptions.SpanId and ActivityTrackingOptions.TraceId](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/logging/?view=aspnetcore-6.0#automatically-log-scope-with-spanid-traceid-parentid-baggage-and-tags) options [that add TraceId and SpanId to the LogScope keys](https://github.com/dotnet/runtime/blob/cb10ed9430762f63a4e9b23a25d142b3ff909879/src/libraries/Microsoft.Extensions.Logging/src/LoggerFactoryScopeProvider.cs#L143). Those keys arrive in the CustomDimension field of the Traces table (which is very convenient). And look how the value of "TraceId" correlates with "OperationId" and "SpanId" with "Id", which makes total sense according to https://docs.microsoft.com/en-us/azure/azure-monitor/app/correlation#correlation-headers-using-w3c-tracecontext:
  - [w3c `parent-id`](https://w3c.github.io/trace-context/#parent-id)=Otel `span-id`=Application Insights `Id`
  - [w3c `trace-id`](https://w3c.github.io/trace-context/#parent-idv)=Application Insights `Operation_Id`
  - [w3c `parent-id`](https://w3c.github.io/trace-context/#parent-id) of this span's parent span=Application Insights `Operation_ParentId`

I am underlining, **this sample is prepared with the standard Microsoft.Extension.Logging**, without OtelCollector at all. This OtelCollector exporter **is just doing same** but explicitly.  

2. TraceId is available as operation id - true. But:
 - There is no Spanid. 
 - Customers prefer to operate (write queries) with the [standard OpenTelemetry terminology (TraceId, SpanId)](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#retrieving-the-traceid-and-spanid) rather than with "Operation Id" and "Id". There are requests on Azure Monitor backlog to bring the terminology to W3C standards at least in UI.
 - Again, adding logging scope (along with TraceId and SpanId) to CustomDimentions is a standard behavior, like it was mentioned above. I didn't invent that, it's not my credit. I just implemented it in this LogExporter too because it's a really convenient feature.